frontend/lib/resolution/default-functions.cpp

```diff
@@ -244,16 +244,15 @@ generateInitSignature(Context* context, const CompositeType* inCompType) {
   for (int i = 0; i < rf.numFields(); i++) {
     auto fieldQt = rf.fieldType(i);
     auto formalName = rf.fieldName(i);
-    bool formalHasDefault = rf.fieldHasDefaultValue(i);
     const uast::Decl* formalAst = nullptr;
+    bool typeHasDefault = isTypeWithDefaultValue(context, fieldQt.type());
+    bool formalHasDefault = rf.fieldHasDefaultValue(i) || typeHasDefault;
```

Comment:
put `typeHasDefault` first in the `||` since we just computed it, so perhaps we don't have to use `fieldHasDefaultvalue` ?